CMakeLists.txt

```diff
@@ -551,47 +507,82 @@ list(REMOVE_ITEM CMAKE_C_FLAGS_MINSIZEREL "-DNDEBUG")
 string(REPLACE ";" " " CMAKE_C_FLAGS_MINSIZEREL "${CMAKE_C_FLAGS_MINSIZEREL}")
 set(CMAKE_C_FLAGS_MINSIZEREL "${CMAKE_C_FLAGS_MINSIZEREL}" CACHE STRING "removed NDEBUG flag" FORCE)
 
-# add pentiumpro arch for windows builds with -O3, otherwise resulting executable may not work
-if(WIN32)
-	separate_arguments(CMAKE_CXX_FLAGS_RELEASE)
-	list(APPEND CMAKE_CXX_FLAGS_RELEASE "-march=pentiumpro")
-	list(REMOVE_DUPLICATES CMAKE_CXX_FLAGS_RELEASE)
-	string(REPLACE ";" " " CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
-	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}" CACHE STRING "added -march=pentiumpro" FORCE)
-	MESSAGE("added -march=pentiumpro to Release build")
-	
-	separate_arguments(CMAKE_C_FLAGS_RELEASE)
-	list(APPEND CMAKE_C_FLAGS_RELEASE "-march=pentiumpro")
-	list(REMOVE_DUPLICATES CMAKE_C_FLAGS_RELEASE)
-	string(REPLACE ";" " " CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
-	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}" CACHE STRING "added -march=pentiumpro" FORCE)
-	MESSAGE("added -march=pentiumpro to Release build")
-	
-	separate_arguments(CMAKE_CXX_FLAGS_OPTIMIZED)
-	list(REMOVE_ITEM CMAKE_CXX_FLAGS_OPTIMIZED "-march=native")
-	list(APPEND CMAKE_CXX_FLAGS_OPTIMIZED "-march=pentiumpro")
-	list(REMOVE_DUPLICATES CMAKE_CXX_FLAGS_OPTIMIZED)
-	string(REPLACE ";" " " CMAKE_CXX_FLAGS_OPTIMIZED
-		"${CMAKE_CXX_FLAGS_OPTIMIZED}")
-	set(CMAKE_CXX_FLAGS_OPTIMIZED "${CMAKE_CXX_FLAGS_OPTIMIZED}" CACHE STRING "added -march=pentiumpro" FORCE)
-	MESSAGE("added -march=pentiumpro to Optimized build")
+# #
+# Start determining options for Release build
+# #
+
+# reset the base Release build option
+MESSAGE("Replacing default flags used for Release build with -O3")
+set(CMAKE_CXX_FLAGS_RELEASE "-O3" CACHE STRING "Release build flags" FORCE)
+set(CMAKE_C_FLAGS_RELEASE "-O3" CACHE STRING "Release build flags" FORCE)
+# set the arch to use for Release build if provided
+if(ARCH)
+	MESSAGE("adding -march=${ARCH} to Release build")
+	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=${ARCH}" CACHE STRING "Release build flags" FORCE)
+	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -march=${ARCH}" CACHE STRING "Release build flags" FORCE)
+endif(ARCH)
+# add pentiumpro arch for windows builds with -O3 if no other arch provided, otherwise resulting executable may not work
+if(WIN32 AND NOT ARCH)
+	MESSAGE("WIN32 and no arch provided, defaulting to pentiumpro")
+	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=pentiumpro" CACHE STRING "Release build flags" FORCE)
+	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -march=pentiumpro" CACHE STRING "Release build flags" FORCE)
+endif(WIN32 AND NOT ARCH)
+
+# if compiling with LTO
+if("${ENABLE_LTO}" STREQUAL "yes")
```

Comment:
Instead of this, please just use

```
if(ENABLE_LTO)
```

Also add

```
option(ENABLE_LTO "Sets Link Time Optimization" OFF)
```

somewhere around line 160.